### ISSUE TITLE
Incorrect IT Securtity and Compliance Standard Investment Report Warnings

### ISSUE THREAD

**Issue by bill-gehrke-hhs** [ombegov/ITDB-schema#220]
As these are only warnings, this is not a critical issue for us during the current submission as we can ignore them.

When submitting multiple ITS&C SIRs in a single XML file, incorrect warnings are generated.  It appear some are due to the validations comparing the data for each SIR to the totals from the IT Investment Portfolio Summary for the first SIR, rather than the current SIR.  As an example:
```
check Submission Successful for UII 009-000000001
check Submission Successful for UII 009-000000002
costsAndCapabilities◦warning The sum of all IT security compliance capabilities for PY ( 2.100000 ) should exactly match Total Security and Compliance Cost PY ( 1.100000 ) from the investment in the IT Portfolio.
◦warning The sum of all IT security compliance capabilities for CY ( 2.200000 ) should exactly match Total Security and Compliance Cost CY ( 1.200000 ) from the investment in the IT Portfolio.
◦warning The sum of all IT security compliance capabilities for BY ( 2.300000 ) should exactly match Total Security and Compliance Cost BY ( 1.300000 ) from the investment in the IT Portfolio.
The sum of capabilities for the PY ( 2.100000 ) should not be greater than the DME and O&M Agency PY Funding and Agency PY Contributions ( 1.100000 ) from the IT Portfolio.
◦warning The sum of capabilities for the CY ( 2.200000 ) should not be greater than the DME and O&M Agency CY Funding and Agency CY Contributions ( 1.200000 ) from the IT Portfolio.
◦warning The sum of capabilities for the BY ( 2.300000 ) should not be greater than the DME and O&M Agency BY Funding and Agency BY Contributions ( 1.300000 ) from the IT Portfolio.
```
In the above, example, no warnings are received for 009-000000001 because the totals match.  However, 009-000000002 receives the above warnings even though $2.1m, $2.2m, and $2.3m are the correct numbers.  The amounts referenced in the warnings - $1.1m, $1.2m, and $1.3m - are the values for 009-000000001.  Note that if I also submit 009-000000003, warnings will continue to reference values from 009-000000001.  

Additionally, the calculations for the 10% warning on "Other" appear not to be reset to 0 for each SIR.  As an example:

```
check Submission Successful for UII 009-000000004
◦warning The amount of spending reported for Other Identify Capabilities (6) for PY ( 2.000000) for this investment should not exceed 10% of the total PY ( 2.000000) for all NIST Framework Category Identify for this investment
check Submission Successful for UII 009-000000005
◦warning The amount of spending reported for Other Identify Capabilities (6) for PY ( 3.000000) for this investment should not exceed 10% of the total PY ( 1.000000) for all NIST Framework Category Identify for this investment
```

In the above example, 009-000000004 has $2m in Other Identify, and it is correctly identified as having the entire $2m of Identify in "Other".  However, 009-000000005 is listed as having $3m in Other Identify, out of a total of $1m in Identify.  This does not make sense as "Other" cannot exceed 100%, and it appears that the issue is that the $3m comes from adding the totals for 009-000000004 and 009-000000005 rather than resetting them and showing $1m, the correct value in this case.  Note that 009-000000006 would continue to add, so if it had $0.5m in other, the warning would be based on $3.5m.

**Comment by ojedanico** [comment]
We have created Jira ticket 1401 to resolve this issue. Once the ticket has been assigned to a sprint I will provide an update with the release date and the solution to fix the issue.

**Comment by ojedanico** [comment]
Ticket 1400 has be added to sprint 2B to be deployed on the 26 Feb 2018.

**Comment by ojedanico** [comment]
This ticket was reprioritized to 3A to be deployed March 12th. 

The fix for this issue is as follow: 

The problem is in src/web/api/Pittsburgh/Controllers/ITDB2/InvestmentReport/validations/IRCostAndCapabilityValidator.php
    protected function getPortfolioTBMTotals()
     {
-        $sqlCacheKey = __METHOD__ . json_encode(func_get_args());
+        $latestAgencySubmittedTimePeriod = ITDB2BaseController::getLatestSubmittedTimePeriodForAgency(InvestmentReport::getAgency());
+        $params = array(BusinessCase::getBudgetYear(),InvestmentReportDetailValidator::$UII, $latestAgencySubmittedTimePeriod);
+
+        $sqlCacheKey = __METHOD__ . json_encode($params);
         if (self::inSQLCache($sqlCacheKey)) {
           return self::lookupSQLCache($sqlCacheKey);
         }
The cache key being generated is not specific to the given UII (it is called for each UII, but the first result is then cached and used for the following UIIs)
My solution is to move the params array to the top to assure a unique cache key
